.github/workflows/sel4bench-hw.yml

```
@@ -0,0 +1,118 @@
+# Copyright 2022, Proofcraft Pty Ltd
+#
+# SPDX-License-Identifier: BSD-2-Clause
+
+# Build and run sel4bench on pull requests, on label request
+
+name: seL4Bench-HW
+
+on:
+  workflow_call:
+
+# intended to run on
+#  pull_request_target:
+#    types: [opened, reopened, synchronize, labeled]
+
+# downgrade permissions to read-only as you would have in a standard PR action
+permissions:
+  contents: read
+
+jobs:
+  code:
+    name: Freeze Code
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'pull_request_target' &&
+               github.event.action != 'labeled' &&
+               contains(github.event.pull_request.labels.*.name, 'hw-bench') ||
+            github.event_name == 'pull_request_target' &&
+               github.event.action == 'labeled' &&
+               github.event.label.name == 'hw-bench' }}
+    outputs:
+      xml: ${{ steps.repo.outputs.xml }}
+    steps:
+    - id: repo
+      uses: seL4/ci-actions/repo-checkout@master
+      with:
+        manifest_repo: sel4bench-manifest
+        manifest: master.xml
+        sha: ${{ github.event.pull_request.head.sha }}
+
+  build:
+    name: Build
+    needs: code
+    runs-on: ubuntu-latest
+    # To reduce the load on GitHub runner numbers and machine queue we cancel
+    # any older runs of this workflow for the current PR.
+    concurrency:
+      group: ${{ github.workflow }}-sel4bench-pr-build-${{ github.event.number }}
+      cancel-in-progress: true
+    strategy:
+      fail-fast: false
+      matrix:
+        march: [armv7a, armv8a, nehalem, rv64imac]
+    steps:
+    - name: Build
+      uses: seL4/ci-actions/sel4bench@master
+      with:
+        xml: ${{ needs.code.outputs.xml }}
+        march: ${{ matrix.march }}
+    - name: Upload images
+      uses: actions/upload-artifact@v4
+      with:
+        name: images-${{ matrix.march }}
+        path: '*-images.tar.gz'
+
+  hw-run:
+    name: HW Benchmark
+    if: ${{ github.repository_owner == 'seL4' }}
+    runs-on: ubuntu-latest
+    needs: [build]
+    concurrency:
+      group: ${{ github.workflow }}-sel4bench-pr-hw-run-${{ github.event.number }}
```

We have to add `strategy.job-index` here again, otherwise the matrix runs kill their silibings.
```suggestion
      group: ${{ github.workflow }}-sel4bench-hw-run-pr-${{ github.event.number }}-${{ strategy.job-index }}
```

---

Actually, even this might not be enough if we are planning to have multiple repos calling this, where the workflow has the same name. We have to even add the `<org>/<repo>` here also to avoid the corner killing case where this is triggered by two PRs from different repos which happen to have the same ID.

---

Concurrency groups have [`repository` scope](https://github.com/orgs/community/discussions/78332), so we don't need to add that one at least. But we do need the job-index as you have it above.

---

Right, and I forgot we are executing in the caller's repo context here (and that's why relative paths here don't work either....)